README.md
Navy SEALs: Covert Operations Gold Edition - Source Code
--------------------------------------------------------

Navy SEALs: Covert Operations is a scale realism total 
conversion for id Software's Quake3/Team Arena engine. 
NS:CO pits players in realistic SEAL and Tango land covert 
operations in realistic settings with scale weapons, 
munitions, and equipment.

NS:CO is a tactically sophisticated feature rich multi-player 
game that emphasized teamwork and realistic SEAL SPECWAR 
missions and tactics.

NS:CO is a total conversion featuring original game-design, 
technology, all original code, stunning art, level design, 
modeling and animation; wrapped in a rich game interface 
that gives players new exciting control over game-play that 
is unique, realistic and enormous fun.

Original Codebase
-----------------

Manfred Nerurkar, Michael Sauer, and id Software.

```
Copyright (C) 1999-2005 Id Software, Inc.
Copyright (C) 2007 Team Mirage

This program is free software; you can redistribute it and/or
modify it under the terms of the GNU General Public License
as published by the Free Software Foundation; either version 2
of the License, or (at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public 
License along with this program; if not, write to the Free 
Software Foundation, Inc., 51 Franklin Street, Fifth Floor, 
Boston, MA 02110-1301, USA.
```



changelog
$Id: changelog,v 1.84 2004/09/12 10:00:45 defconx Exp $

--defconx: start

2006/01/25:
* beautified the entire codebase using 'astyle.sourceforge.net'
* fixed alot of crash bugs
* added new trapcall for extended sound calls
* added new cursor_dn function to the ui
* resized cursor to 64 points
* cleaned up code ( removed alot of ta/q3a functions )
* removed chat rendering from clientgame

--defconx: end

2004/09/12:
* patched g_seals.c
	fix for new menu system (gamestate will only be sent if it has
	changed)

2004/09/07:
* patched g_seals.c
	fix for new menu system (GameState)
* patched ui_main.c
	added ui_gamestate variable
* patched files concerning client/server communication
	server tells client now the gamestate (EV_GAMESTATE)

2004/08/29:
* patched menu files
	new menu system
* patched ui_main.c
	removed XP lag
	new handeling of inventory for character submission
* patched g_seals.c
	new handling in inventory submission (includes character)

2004/08/28:
* patched cg_main.c
	removed a buffer overflow in cg_weapons array
	added com_maxfps protection
	removed defcons nutshell
* patched cg_view.c
	removed defcons nutshell
* patched cg_weapons.c
	removed a buffer overflow in weapon handling
	removed a buffer overflow in weaponOffsets array
	rewrote ammunition stuff
* patched bg_misc.c
	rewrote ammunition stuff
	fixed BG_GotWeapon (defcons nutshell)
* patched bg_pmove.c
	rewrote ammunition stuff
* patched bg_public.h
	removed defcons nutshell
* patched g_active.c
	rewrote ammunition stuff
* patched g_cmds.c
	removed defcons nutshell
* patched g_items.c
	rewrote ammunition stuff
	removed defoncs nutshell
* patched g_misc.c
	rewrote ammunition stuff
* patched g_missile.c
	real handling of 40mm Grenades
	rewrote ammunition stuff
* patched g_mover.c
	fixed soundcode for elevator doors
* patched g_seals.c
	rewrote ammunition stuff (buffer overflow was here)
	removed the ability to scope while in GL mode
	restricted number of grenades to SEALS_MAX_GRENADES / cheat protection
* patched g_weapon.c
	removed ugly code
	rewrote ammunition stuff
* patched variables.h
	introduced SEALS_MAX_GRENADES (default 4)
	rewrote ammunition stuff
* patched ui_local.h
	increased MAX_MENUITEMS from 96 to 256 (was buffer overflow)
* patched ui_main.c
	removed unnessesary code
	added CG_HandleGrenadeButton
* patched ui_shared.h
	increased MAX_MENUITEMS from 96 to 256 (was buffer overflow)
* patched menudef.h
	introduced new grenade selection system
	fixed weapon numbers (removed defcons nutshell)

2004/08/24:
* patched g_mover.c 
	removed the bug with door rotating (soundbug)
* patched variables.h
	tweaked the GL
* patched g_seals.c (weaponmode)
	no scope while using the GL

2004/08/23:
* patched g_mover.c
	added door-sounds for elevator-doors
* patched g_trigger.c
	made the trigger_toggle trigger only for one team (usable
	with team value)

2004/08/18:
* patched cg_view.c
	removed the smokegren bug (when picking up briefcase)
	look for WP_SMOKE - 1
* patched cg_weapons.c
	tweaked the flashlight, it's not tooooo bad now, however
	the visual sux, but they do the job (blame q3)

2004/08/15:
* patched lots of files
	func_elevator_button spawns a message when pointed on it
	new trigger: trigger_use
	tweaked smokegrenades
	tweaked GL
* patched g_spawn.c g_trigger.c
	added trigger_toggle

2004/08/11:
* patched g_weapon.c
	bullets lose damage when penetrating a wall
	wall penetration tweaked (walls must be thin to let the client pass)
	fixed bug in Fire_Lead (one was able to shoot himself in some very
	unlikely circumstances)
* patched cg_newDraw.c
	no crosshair for m4/ak47 when in GL mode
	smokegrenades to 1/2 secs

2004/08/10:
* patched g_missile g_combat g_weapons ui_main ui_shared g_seals
	added hand-throwable smokegrenades
	added old GL
	fixed but in Fire_Lead

2004/08/09:
* patched g_missile.c g_combat.c g_weapons.c
	added support for shootable grenades

2004/08/08:
* patched g_mover.c
	fixed last bug with elevator code, now it works fine

2004/08/07:
* patched g_gameentities.c g_mover.c
	included func_elevator and func_elevator_button, elevator doors
	work with func_door

2004/08/06:
* patched g_unlag.c g_weapon.c g_active.c g_main.c g_local.h
	reincluded full antilag, but one can disable it with the server
	cvar g_antilag (default is 1)
* patched bg_pmove.c
	fixed the stamina consumption bug
	sprinting lasts now longer
	stamina damage for jumping
* patched g_combat.c
	rewrote G_Damage lots of stuff
	stamina damage for grenades
* patched variables.h
	range of grenades is now bigger
	kevlar penetration variables for every weapon

2004/08/01:
* patched g_unlag.c g_seals.h g_weapon.c g_active.c
	reincluded unlag, but only with 50% effect
* patched the above files
	removed antilag again (sux)
* patched g_active.c
	made the falldamage a lil bit higher
* Houston, we have a problem:
	the player hit box is not only used for hit-detection, but also for
	hit-detection with walls. lowering it to fix the headshotzone will break
	the hit-detection with the world (players are able to 'look' through walls.
	what to do now? we definitly have to fix the hitboxes :|
	Actually we would have to switch to hitboxes everytime, we make a trace to
	see if a weapon hits a player. Meaning go through all entities, if it's a 
	player entity, fix hitboxes, make the trace, unfix hitboxes.
	It's a hack, but at the moment the only possible solution in my mind
* Houston, problem solved
* patched g_weapons.c g_seals.h g_seals.c
	added functions NS_FixHitboxes and NS_RestoreHitboxes. for weapon hits just
	put NS_FixHitboxes in front of the trace and NS_RestoreHitboxes after the
	trace.

2004/07/31:
* patched g_combat.c g_seals.c
	fixed falling damage
* patched bg_pmove.c
	fixed falling damage
* patched g_combat.c g_seals.c bg_pmove.c (search for MAXS_Z)
	fixed hitzones

2004/07/20:
* patched game/g_unlag.c and all related files
	removed unlagged code

2004/05/15:
* added docs/dd_195.tex
	i started writing a design doc for ns:co 1.95

2004/04/04:

ui/menudef.h
/*
===========================================================================
Copyright (C) 1999-2005 Id Software, Inc.
Copyright (C) 2007 Team Mirage

This file is part of Navy SEALs: Covert Operations source code.

Navy SEALs: Covert Operations source code is free software; you can
redistribute it and/or modify it under the terms of the GNU General Public
License as published by the Free Software Foundation; either version 2 of
the License, or (at your option) any later version.

Navy SEALs: Covert Operations source code is distributed in the hope that it
will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty
of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with Navy SEALs: Covert Operations source code; if not, write to the
Free Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
02110-1301  USA
===========================================================================
*/

#define ITEM_TYPE_TEXT 0                  // simple text
#define ITEM_TYPE_BUTTON 1                // button, basically text with a border 
#define ITEM_TYPE_RADIOBUTTON 2           // toggle button, may be grouped 
#define ITEM_TYPE_CHECKBOX 3              // check box
#define ITEM_TYPE_EDITFIELD 4             // editable text, associated with a cvar
#define ITEM_TYPE_COMBO 5                 // drop down list
#define ITEM_TYPE_LISTBOX 6               // scrollable list  
#define ITEM_TYPE_MODEL 7                 // model
#define ITEM_TYPE_OWNERDRAW 8             // owner draw, name specs what it is
#define ITEM_TYPE_NUMERICFIELD 9          // editable text, associated with a cvar
#define ITEM_TYPE_SLIDER 10               // mouse speed, volume, etc.
#define ITEM_TYPE_YESNO 11                // yes no cvar setting
#define ITEM_TYPE_MULTI 12                // multiple list setting, enumerated
#define ITEM_TYPE_BIND 13		          // multiple list setting, enumerated
    
#define ITEM_ALIGN_LEFT 0                 // left alignment
#define ITEM_ALIGN_CENTER 1               // center alignment
#define ITEM_ALIGN_RIGHT 2                // right alignment

#define ITEM_TEXTSTYLE_NORMAL 0           // normal text
#define ITEM_TEXTSTYLE_BLINK 1            // fast blinking
#define ITEM_TEXTSTYLE_PULSE 2            // slow pulsing
#define ITEM_TEXTSTYLE_SHADOWED 3         // drop shadow ( need a color for this )
#define ITEM_TEXTSTYLE_OUTLINED 4         // drop shadow ( need a color for this )
#define ITEM_TEXTSTYLE_OUTLINESHADOWED 5  // drop shadow ( need a color for this )
#define ITEM_TEXTSTYLE_SHADOWEDMORE 6     // drop shadow ( need a color for this )
                          
#define WINDOW_BORDER_NONE 0              // no border
#define WINDOW_BORDER_FULL 1              // full border based on border color ( single pixel )
#define WINDOW_BORDER_HORZ 2              // horizontal borders only
#define WINDOW_BORDER_VERT 3              // vertical borders only 
#define WINDOW_BORDER_KCGRADIENT 4        // horizontal border using the gradient bars
  
#define WINDOW_STYLE_EMPTY 0              // no background
#define WINDOW_STYLE_FILLED 1             // filled with background color
#define WINDOW_STYLE_GRADIENT 2           // gradient bar based on background color 
#define WINDOW_STYLE_SHADER   3           // gradient bar based on background color 
#define WINDOW_STYLE_TEAMCOLOR 4          // team color
#define WINDOW_STYLE_CINEMATIC 5          // cinematic

#define MENU_TRUE 1                       // uh.. true
#define MENU_FALSE 0                      // and false

#define HUD_VERTICAL				0x00
#define HUD_HORIZONTAL				0x01

// list box element types
#define LISTBOX_TEXT  0x00
#define LISTBOX_IMAGE 0x01

// list feeders
#define FEEDER_HEADS						0x00			// model heads
#define FEEDER_MAPS							0x01			// text maps based on game type
#define FEEDER_SERVERS						0x02			// servers
#define FEEDER_CLANS						0x03			// clan names
#define FEEDER_ALLMAPS						0x04			// all maps available, in graphic format
#define FEEDER_REDTEAM_LIST					0x05			// red team members
#define FEEDER_BLUETEAM_LIST				0x06			// blue team members
#define FEEDER_PLAYER_LIST					0x07			// players
#define FEEDER_TEAM_LIST					0x08			// team members for team voting
#define FEEDER_MODS							0x09			// team members for team voting
#define FEEDER_DEMOS 						0x0a			// team members for team voting
#define FEEDER_SCOREBOARD					0x0b			// team members for team voting
#define FEEDER_Q3HEADS		 				0x0c			// model heads
#define FEEDER_SERVERSTATUS					0x0d			// server status
#define FEEDER_FINDPLAYER					0x0e			// find player
#define FEEDER_CINEMATICS					0x0f			// cinematics
#define FEEDER_SCRIPTS						0x10			// scripts in nssl/

// display flags
#define CG_SHOW_BLUE_TEAM_HAS_REDFLAG     0x00000001
#define CG_SHOW_RED_TEAM_HAS_BLUEFLAG     0x00000002
#define CG_SHOW_ANYTEAMGAME               0x00000004
#define CG_SHOW_HARVESTER                 0x00000008
#define CG_SHOW_ONEFLAG                   0x00000010
#define CG_SHOW_CTF                       0x00000020
#define CG_SHOW_OBELISK                   0x00000040
#define CG_SHOW_HEALTHCRITICAL            0x00000080
#define CG_SHOW_SINGLEPLAYER              0x00000100
#define CG_SHOW_TOURNAMENT                0x00000200
#define CG_SHOW_DURINGINCOMINGVOICE       0x00000400
#define CG_SHOW_IF_PLAYER_HAS_FLAG	  0x00000800
#define CG_SHOW_LANPLAYONLY		  0x00001000
#define CG_SHOW_MINED			  0x00002000
#define CG_SHOW_HEALTHOK		  0x00004000
#define CG_SHOW_TEAMINFO		  0x00008000
#define CG_SHOW_NOTEAMINFO		  0x00010000
#define CG_SHOW_OTHERTEAMHASFLAG	  0x00020000
#define CG_SHOW_YOURTEAMHASENEMYFLAG	  0x00040000
#define CG_SHOW_ANYNONTEAMGAME		  0x00080000
#define CG_SHOW_2DONLY			  0x10000000
#define CG_SHOW_SPECTATOR		  0x20000000
#define CG_SHOW_NO_SPECTATOR		  0x40000000

#define UI_SHOW_LEADER				                0x00000001
#define UI_SHOW_NOTLEADER				        0x00000002
#define UI_SHOW_FAVORITESERVERS					0x00000004
#define UI_SHOW_ANYNONTEAMGAME					0x00000008
#define UI_SHOW_ANYTEAMGAME					0x00000010
#define UI_SHOW_NEWHIGHSCORE					0x00000020
#define UI_SHOW_DEMOAVAILABLE					0x00000040
#define UI_SHOW_NEWBESTTIME					0x00000080
#define UI_SHOW_FFA						0x00000100
#define UI_SHOW_NOTFFA						0x00000200
#define UI_SHOW_NETANYNONTEAMGAME	 			0x00000400
#define UI_SHOW_NETANYTEAMGAME		 			0x00000800
#define UI_SHOW_NOTFAVORITESERVERS				0x00001000 
 




// owner draw types
// ideally these should be done outside of this file but
// this makes it much easier for the macro expansion to 
// convert them for the designers ( from the .menu files )
#define CG_OWNERDRAW_BASE			1

#define CG_PLAYER_ROUNDS_ICON		1              
#define CG_PLAYER_ROUNDS_VALUE		2

#define CG_PLAYER_HEAD				3
#define CG_PLAYER_HEALTH			4

#define CG_PLAYER_AMMO_ICON			5
#define CG_PLAYER_AMMO_VALUE		6

#define CG_SELECTEDPLAYER_HEAD		7

#define CG_SELECTEDPLAYER_NAME		8
#define CG_SELECTEDPLAYER_LOCATION	9
#define CG_SELECTEDPLAYER_STATUS	10
#define CG_SELECTEDPLAYER_WEAPON	11
#define CG_SELECTEDPLAYER_POWERUP	12

#define CG_FLAGCARRIER_HEAD 13
#define CG_FLAGCARRIER_NAME 14
#define CG_FLAGCARRIER_LOCATION 15
#define CG_FLAGCARRIER_STATUS 16
#define CG_FLAGCARRIER_WEAPON 17
#define CG_FLAGCARRIER_POWERUP 18

#define CG_PLAYER_ITEM 19
#define CG_PLAYER_SCORE 20

#define CG_BLUE_FLAGHEAD 21
#define CG_BLUE_FLAGSTATUS 22
#define CG_BLUE_FLAGNAME 23
#define CG_RED_FLAGHEAD 24
#define CG_RED_FLAGSTATUS 25
#define CG_RED_FLAGNAME 26

#define CG_BLUE_SCORE 27
#define CG_RED_SCORE 28
#define CG_RED_NAME 29
#define CG_BLUE_NAME 30
#define CG_HARVESTER_SKULLS 31					// only shows in harvester
#define CG_ONEFLAG_STATUS 32						// only shows in one flag
#define CG_PLAYER_LOCATION 33
#define CG_TEAM_COLOR 34
#define CG_CTF_POWERUP 35
                                        
#define CG_AREA_POWERUP	36
#define CG_AREA_LAGOMETER	37            // painted with old system
#define CG_PLAYER_HASFLAG 38            
#define CG_GAME_TYPE 39                 // not done

#define CG_SELECTEDPLAYER_ARMOR 40      
#define CG_SELECTEDPLAYER_HEALTH 41
#define CG_PLAYER_STATUS 42
#define CG_FRAGGED_MSG 43               // painted with old system
#define CG_PROXMINED_MSG 44             // painted with old system
#define CG_AREA_FPSINFO 45              // painted with old system
#define CG_AREA_SYSTEMCHAT 46           // painted with old system
#define CG_AREA_TEAMCHAT 47             // painted with old system
#define CG_AREA_CHAT 48                 // painted with old system
#define CG_GAME_STATUS 49
#define CG_KILLER 50
#define CG_PLAYER_ARMOR_ICON2D 51              
#define CG_PLAYER_AMMO_ICON2D 52
#define CG_ACCURACY 53
#define CG_ASSISTS 54
#define CG_DEFEND 55
#define CG_EXCELLENT 56
#define CG_IMPRESSIVE 57
#define CG_PERFECT 58
#define CG_GAUNTLET 59
#define CG_SPECTATORS 60
#define CG_TEAMINFO 61
#define CG_VOICE_HEAD 62
#define CG_VOICE_NAME 63
#define CG_PLAYER_HASFLAG2D 64            
#define CG_HARVESTER_SKULLS2D 65					// only shows in harvester
#define CG_CAPFRAGLIMIT 66	 
#define CG_1STPLACE 67
#define CG_2NDPLACE 68
#define CG_CAPTURES 69

#define UI_OWNERDRAW_BASE 200
#define UI_HANDICAP 200
#define UI_EFFECTS 201
#define UI_PLAYERMODEL 202
#define UI_CLANNAME 203
#define UI_CLANLOGO 204
#define UI_GAMETYPE 205
#define UI_MAPPREVIEW 206
#define UI_SKILL 207
#define UI_BLUETEAMNAME 208
#define UI_REDTEAMNAME 209
#define UI_NETSOURCE 220
#define UI_NETMAPPREVIEW 221
#define UI_NETFILTER 222
#define UI_TIER 223
#define UI_OPPONENTMODEL 224
#define UI_TIERMAP1 225
#define UI_TIERMAP2 226
#define UI_TIERMAP3 227
#define UI_PLAYERLOGO 228
#define UI_OPPONENTLOGO 229
#define UI_PLAYERLOGO_METAL 230
#define UI_OPPONENTLOGO_METAL 231
#define UI_PLAYERLOGO_NAME 232
#define UI_OPPONENTLOGO_NAME 233
#define UI_TIER_MAPNAME 234
#define UI_TIER_GAMETYPE 235
#define UI_ALLMAPS_SELECTION 236
#define UI_OPPONENT_NAME 237
#define UI_VOTE_KICK 238
#define UI_BOTNAME 239
#define UI_BOTSKILL 240
#define UI_REDBLUE 241
#define UI_CROSSHAIR 242
#define UI_SELECTEDPLAYER 243
#define UI_MAPCINEMATIC 244
#define UI_NETGAMETYPE 245
#define UI_NETMAPCINEMATIC 246
#define UI_SERVERREFRESHDATE 247
#define UI_SERVERMOTD 248
#define UI_GLINFO  249
#define UI_KEYBINDSTATUS 250
#define UI_CLANCINEMATIC 251
#define UI_MAP_TIMETOBEAT 252
#define UI_JOINGAMETYPE 253
#define UI_PREVIEWCINEMATIC 254
#define UI_STARTMAPCINEMATIC 255
#define UI_MAPS_SELECTION 256
#define CG_PICKUP 257
#define CG_PRIMARY	258
#define CG_SECONDARY	259
#define CG_CHARACTERBUTTON 260
#define UI_SEAL_HEADMODEL 261
#define UI_TANGO_HEADMODEL 262
#define UI_SEAL_PLAYERMODEL 263
#define UI_TANGO_PLAYERMODEL 264

// height scalable, rounds bar, drawed using bullet graphics
#define CG_PLAYER_CLIPGFX	265
// value for clipgfx this!
#define CG_PLAYER_CLIPVALUE	266

// damage locator , scalable
#define CG_PLAYER_DAMAGELOC	267
// rounds bar hor, scalable
#define CG_PLAYER_AMMOBAR_HOR	268
// rounds bar vert, scalable
#define CG_PLAYER_AMMOBAR_VERT	269

// clip icon ( grenade/rifle/shotgun etc...), scalable
#define CG_PLAYER_CLIPS_ICON	270
// clip value amount of clips left draw using digits, scalable
#define CG_PLAYER_CLIPS_VALUE	271

#define CG_TIMER_WORLD		272 // world timer, roundtime/sp server time
#define CG_TIMER_VIP		273 // vip timer
#define CG_TIMER_ASSAULT1	274 // assault timer (more required!!!!!!!!!!!!!!!!!!!!!)
#define CG_TIMER_ASSAULT2   295
#define CG_TIMER_ASSAULT3   296
#define CG_TIMER_ASSAULT4   297
#define CG_TIMER_BOMB		298


// stamina bar, no border, supply with shader!, both scalable
#define CG_PLAYER_STAMINABAR_HOR	275
#define CG_PLAYER_STAMINABAR_VERT	276

#define CG_PLAYER_WEAPONSTATUS		277

#define UI_PRIMARY_ADDON1		278
#define UI_PRIMARY_ADDON2		279
#define UI_PRIMARY_ADDON3		280

#define UI_SIDEARM_ADDON1		281
#define UI_SIDEARM_ADDON2		282

#define UI_TOTALWEIGHT			283

#define UI_TEXT_PRIMARY			284
#define UI_TEXT_SIDEARM			285

#define CG_TEXT_PICKUP			286 // display not pickup icon /do text
 

#define CG_CHARACTER_STRENGTH		287
#define CG_CHARACTER_TECHINCAL		288
#define CG_CHARACTER_STAMINA		289
#define CG_CHARACTER_ACCURACY		290
#define CG_CHARACTER_SPEED		291
#define CG_CHARACTER_STEALTH		292

#define UI_BLUETEAM1 300
#define UI_BLUETEAM2 301
#define UI_BLUETEAM3 302
#define UI_BLUETEAM4 303
#define UI_BLUETEAM5 304
#define UI_BLUETEAM6 305
#define UI_REDTEAM1 306
#define UI_REDTEAM2 307
#define UI_REDTEAM3 308
#define UI_REDTEAM4 309
#define UI_REDTEAM5 310
#define UI_REDTEAM6	311

#define CG_PLAYER_ENEMY 322

#define CG_INVEN_AMMO_FLASH 330
#define CG_INVEN_AMMO_SMOKE 331
#define CG_INVEN_AMMO_MK26  332
#define CG_INVEN_AMMO_40MM  333

// list delimit at: 311
 

// do not change these defines! they are synced with the gamecode ( both, CGAME & GAME )

#ifndef CGAME
#define WP_NONE			0
#define WP_KHURKURI		1
#define	WP_SEALKNIFE	2
#define WP_C4			3
#define WP_GRENADE		4
#define	WP_FLASHBANG	5
#define WP_MK23			6
#define WP_GLOCK		7
#define	WP_SW40T		8
#define	WP_P9S			9
#define WP_DEAGLE   10
#define WP_SW629    11
#define	WP_PDW			12
#define WP_MAC10		13
#define WP_MP5			14
#define	WP_AK47			15
#define	WP_M4			  16
#define	WP_PSG1			17
#define WP_MACMILLAN	18
#define	WP_870			19
#define	WP_M590			20
#define WP_SPAS15		21
#define WP_M14      22
#define WP_M249     23
#define WP_SL8SD    24
#define WP_SMOKE    25
#define WP_NUM_WEAPONS	26
#endif


// these defines are important for the 'setItem' uiScript
// items are toggable.
// if you want to remove all weaponitems at once use the
// uiScript 'removeWeaponItems'
// set as ownerParam
// 
// armor 
//
#define ITEM_HELMET					1
#define ITEM_VEST					2
// 
// weapon equipment
//
#define ITEM_SILENCER				3
#define ITEM_LASERSIGHT				4
#define ITEM_GRENADELAUNCHER		5
#define ITEM_BAYONET				6
#define ITEM_SCOPE					7 
#define ITEM_SILENCER_SECONDARY		8
#define ITEM_LASERSIGHT_SECONDARY			9
#define ITEM_GRENADELAUNCHER_SECONDARY		10
#define ITEM_BAYONET_SECONDARY				11
#define ITEM_SCOPE_SECONDARY				12
#define ITEM_DUCKBILL						13
#define ITEM_FLASHLIGHT						14
#define ITEM_FLASHLIGHT_SECONDARY			15 // not implemented yet.

//
// player character
//  ownerParm flags for the 'raiseCharacter' uiScript
// and for their responsible buttons... 
// they should get the same ownerParam
// and it should be ownerdraw CG_CHARACTERBUTTON
// done
#define PC_STRENGTH		1
// technical
#define PC_TECHNICAL		2
// done
#define PC_STAMINA		3
// done
#define PC_ACCURACY		4
// done
#define PC_SPEED		5
// done
#define PC_STEALTH		6 

//
// todo:
//
//  -6 buttons with ownerdraw CG_CHARACTERBUTTON and ownerParam 
//        ownerparm = if the button should change stamina, add as
//                      ownerParm PC_STAMINA
//        ownerDraw = CG_CHARACTERBUTTON
//     **this button will then turn red if there are not enough XP
//     ** and green if it's possible
//       ** set as 'action' { uiScript "raiseCharacter" }
//
//  -to show current XP use cvarshow "char_xp"  ( same as ammo show )
//
//

// cvar to show xperience points: "char_xp"

#define VOICECHAT_GETFLAG			"getflag"				// command someone to get the flag
#define VOICECHAT_OFFENSE			"offense"				// command someone to go on offense
#define VOICECHAT_DEFEND			"defend"				// command someone to go on defense
#define VOICECHAT_DEFENDFLAG		"defendflag"			// command someone to defend the flag
#define VOICECHAT_PATROL			"patrol"				// command someone to go on patrol (roam)
#define VOICECHAT_CAMP				"camp"					// command someone to camp (we don't have sounds for this one)
#define VOICECHAT_FOLLOWME			"followme"				// command someone to follow you
#define VOICECHAT_RETURNFLAG		"returnflag"			// command someone to return our flag
#define VOICECHAT_FOLLOWFLAGCARRIER	"followflagcarrier"		// command someone to follow the flag carrier
#define VOICECHAT_YES				"yes"					// yes, affirmative, etc.
#define VOICECHAT_NO				"no"					// no, negative, etc.
#define VOICECHAT_ONGETFLAG			"ongetflag"				// I'm getting the flag
#define VOICECHAT_ONOFFENSE			"onoffense"				// I'm on offense
#define VOICECHAT_ONDEFENSE			"ondefense"				// I'm on defense
#define VOICECHAT_ONPATROL			"onpatrol"				// I'm on patrol (roaming)
#define VOICECHAT_ONCAMPING			"oncamp"				// I'm camping somewhere
#define VOICECHAT_ONFOLLOW			"onfollow"				// I'm following
#define VOICECHAT_ONFOLLOWCARRIER	"onfollowcarrier"		// I'm following the flag carrier
#define VOICECHAT_ONRETURNFLAG		"onreturnflag"			// I'm returning our flag
#define VOICECHAT_INPOSITION		"inposition"			// I'm in position
#define VOICECHAT_IHAVEFLAG			"ihaveflag"				// I have the flag
#define VOICECHAT_BASEATTACK		"baseattack"			// the base is under attack
#define VOICECHAT_ENEMYHASFLAG		"enemyhasflag"			// the enemy has our flag (CTF)
#define VOICECHAT_STARTLEADER		"startleader"			// I'm the leader
#define VOICECHAT_STOPLEADER		"stopleader"			// I resign leadership
#define VOICECHAT_TRASH				"trash"					// lots of trash talk
#define VOICECHAT_WHOISLEADER		"whoisleader"			// who is the team leader
#define VOICECHAT_WANTONDEFENSE		"wantondefense"			// I want to be on defense
#define VOICECHAT_WANTONOFFENSE		"wantonoffense"			// I want to be on offense
#define VOICECHAT_KILLINSULT		"kill_insult"			// I just killed you
#define VOICECHAT_TAUNT				"taunt"					// I want to taunt you
#define VOICECHAT_DEATHINSULT		"death_insult"			// you just killed me
#define VOICECHAT_KILLGAUNTLET		"kill_gauntlet"			// I just killed you with the gauntlet
#define VOICECHAT_PRAISE			"praise"				// you did something good
